frontend/catalyst/jax_primitives.py

```diff
@@ -551,13 +559,87 @@ def _apply_registered_pass_lowering(
 
 
 #
-# func
+# module
 #
-mlir_fn_cache: Dict["catalyst.jax_tracer.Function", Any] = {}
+
+mlir_fn_cache: Dict[Any, Any] = {}
+
+
+@module_p.def_impl
+def _module_def_impl(*args, call_jaxpr, fn):  # pragma: no cover
+    raise NotImplementedError()
+
+
+def _module_to_mlir(ctx, *args, call_jaxpr, fn, call=True):
+
+    if not isinstance(fn, qml.QNode) and not call:
+        _func_lowering(ctx, *args, call_jaxpr=call_jaxpr, fn=fn, call=call)
+        return None
+
+    output_types = list(map(mlir.aval_to_ir_types, ctx.avals_out))
+    flat_output_types = util.flatten(output_types)
+
+    if mlir_fn_cache.get(fn) and not call:
+        return None
+
+    symbol_table = ctx.module_context.symbol_table
+    if mlir_fn_cache.get(fn):
+        func_op = mlir_fn_cache[fn]
+        parent = func_op.parent.operation.attributes["sym_name"].value
+        symbol_name = ir.SymbolRefAttr.get([parent, func_op.name.value])
+        call = CallNestedModuleOp(
+            flat_output_types,
+            symbol_name,
+            mlir.flatten_lowering_ir_args(args),
+        )
+
+        return call.results
+
+    ctx.allow_unregistered_dialects = True
+    root = ctx.module_context.module
+    with ir.InsertionPoint(root.body):
+        module = ModuleOp()
+        symbol_attr = ir._symbolNameAttr("module_" + fn.__name__, ctx.module_context.context)
+        module.operation.attributes["sym_name"] = symbol_attr
+        symbol_table.insert(module)
+    nrep = jaxpr_replicas(call_jaxpr)
+    ctx_params = {}
```

Comment:
Why are we hardcoding these parameters instead of just retrieving them from `ctx.module_context`? Shouldn't the nested qnode modules have the same parameters as the big parent module?

Reply:
Oh I saw your TODO in PR description, never mind

Reply:
See here: https://github.com/PennyLaneAI/catalyst/pull/1144/commits/44b84d40c9c9801b5b9d4d90e5b43d172dbc9619